lib/cli/run-helpers.js

```diff
@@ -0,0 +1,335 @@
+};
+
+/**
+ * Dump list of built-in reporters
+ * @private
+ */
+exports.runShowReporters = () => {
+  showKeys(Mocha.reporters);
+};
+
+/**
+ * Process extensions for `--watch`, `--compilers`, etc.
+ * @param {yargs.Arguments} argv - Arguments
+ * @private
+ */
+exports.handleExtensions = argv => {
+  const compilers = argv.compilers || [];
+  if (compilers.length) {
+    require('util').deprecate(() => {},
+    '"--compilers" will be removed in a future version of Mocha; see https://git.io/vdcSr for more info')();
+  }
+
+  argv.extensions = ['js'];
+  compilers.forEach(c => {
+    const idx = c.indexOf(':');
+    const ext = c.slice(0, idx);
+    let mod = c.slice(idx + 1);
+
+    if (mod[0] === '.') {
+      mod = path.join(cwd, mod);
+    }
+    require(mod);
+    argv.extensions.push(ext);
+    argv.watchExtensions.push(ext);
+  });
+};
+
+/**
+ * `require()` the modules as required by `--require <require>`
+ * @param {yargs.Arguments} argv - Arguments
+ * @private
+ */
+exports.handleRequires = argv => {
+  argv.require.forEach(mod => {
+    let modpath = mod;
+    if (fs.existsSync(mod, {cwd}) || fs.existsSync(`${mod}.js`, {cwd})) {
+      modpath = path.resolve(mod);
+      debug(`resolved ${mod} to ${modpath}`);
+    }
+    require(modpath);
+  });
+};
+
+/**
+ * Smash together an array of test files in the correct order
+ * @param {yargs.Arguments} argv - Arguments
+ * @returns {string[]} List of files to test
+ * @private
+ */
+exports.handleFiles = argv => {
+  let files = [];
+  argv.spec.forEach(arg => {
+    let newFiles;
+    try {
+      newFiles = utils.lookupFiles(arg, argv.extensions, argv.recursive);
+    } catch (err) {
+      if (err.message.indexOf('cannot resolve path') === 0) {
+        console.error(
+          `Warning: Could not find any test files matching pattern: ${arg}`
+        );
+        return;
+      }
+
+      throw err;
+    }
+
+    if (typeof newFiles !== 'undefined') {
+      if (typeof newFiles === 'string') {
+        newFiles = [newFiles];
+      }
+      newFiles = newFiles.filter(fileName =>
+        (argv.exclude || []).every(pattern => !minimatch(fileName, pattern))
+      );
+    }
+
+    files = files.concat(newFiles);
+  });
+
+  if (!files.length) {
+    console.error(chalk.red(`${symbols.error} No test files found`));
+    process.exit(1);
+  }
+
+  const fileArgs = (argv.file || []).map(filepath => path.resolve(filepath));
+  files = files.map(filepath => path.resolve(filepath));
+
+  // ensure we don't sort the stuff from fileArgs; order is important!
+  if (argv.sort) {
+    files.sort();
+  }
+
+  // add files given through --file to be ran first
+  files = fileArgs.concat(files);
+  debug('files (in order): ', files);
+  return files;
+};
+
+/**
+ * Actually run tests
+ * @param {Mocha} mocha - Mocha instance
+ * @param {yargs.Arguments} argv - Arguments
+ * @param {string[]} files - Array of test files
+ * @private
+ */
+exports.runMocha = (mocha, argv, files) => {
+  let runner;
```

Comment:
Want to consider adding requested changes from #1810 here? Or is it just too much with all the rest of this?